sklearn/preprocessing/label.py

```diff
@@ -54,69 +57,117 @@ def _check_numpy_unicode_bug(labels):
+    estimator : object, (default=LinearRegression())
+        Estimator object derived from ``RegressorMixin``.
+
+    transformer : object, (default=None)
+        Estimator object derived from ``TransformerMixin``. Cannot be set at
+        the same time as ``func`` and ``inverse_func``. If ``None`` and
+        ``func`` and ``inverse_func`` as well, the transformer will be an
+        identity transformer.
 
-    Parameter
-    ---------
-    estimator : object
-        Estimator object to wrap.
+    func : function, (default=None)
+        Function to apply to ``y`` before passing to ``fit``. Cannot be set at
+        the same time than ``transformer``. If ``None`` and ``transformer`` as
+        well, the function used will be the identity function.
 
-    func : function
-        Function to apply to y before passing to fit.
+    inverse_func : function, (default=None)
+        Function apply to the prediction of the estimator. Cannot be set at
+        the same time than ``transformer``. If ``None`` and ``transformer`` as
+        well, the function used will be the identity function.
 
-    inverse_func : function
-        Function apply to the prediction of the estimator.
+    check_invertible : bool, (default=True)
+        Whether to check that ``transform`` followed by ``inverse_transform``
+        or ``func`` followed by ``inverse_func`` lead to the original data.
 
     Attributes
     ----------
     estimator_ : object
         Fitted estimator.
 
+    transformer_ : object
+        Used transformer in ``fit`` and ``predict``.
+
     Examples
     --------
     >>> import numpy as np
     >>> from sklearn.linear_model import LinearRegression
-    >>> from sklearn.preprocessing.label import TargetTransformer
-    >>> tt = TargetTransformer(LinearRegression(), func=np.log,
-    ...                        inverse_func=np.exp)
+    >>> from sklearn.preprocessing.label import TransformedTargetRegressor
+    >>> tt = TransformedTargetRegressor(estimator=LinearRegression(),
+    ...                                 func=np.log, inverse_func=np.exp)
     >>> X = np.arange(4).reshape(-1, 1)
     >>> y = np.exp(2 * X).ravel()
     >>> tt.fit(X, y)
     ... #doctest: +NORMALIZE_WHITESPACE
-    TargetTransformer(estimator=LinearRegression(copy_X=True,
-        fit_intercept=True, n_jobs=1, normalize=False), func=<ufunc 'log'>,
-        inverse_func=<ufunc 'exp'>)
+    TransformedTargetRegressor(check_invertible=True,
+        estimator=LinearRegression(copy_X=True, fit_intercept=True, n_jobs=1,
+                                   normalize=False),
+        func=<ufunc 'log'>, inverse_func=<ufunc 'exp'>, transformer=None)
     >>> tt.score(X, y)
     1.0
     >>> tt.estimator_.coef_
     array([ 2.])
 
-
-    Notes
-    -----
-    It is not checked whether func and inverse_func are actually
-    inverse to each other.
     """
-    def __init__(self, estimator, func, inverse_func):
+    def __init__(self, estimator, transformer=None,
```

Comment:
Didn't you say LinearRegression is the default estimator? That might be good for now before we got better common tests.